projects/app-lob/src/app/grid-dynamic-chart-data/data-analysis-dock-manager/data-analysis-dock-manager.component.ts

```diff
@@ -96,6 +103,30 @@ export class DataAnalysisDockManagerComponent implements OnInit, AfterViewInit {
                 chart.indexOf('Bar') === -1 ||
                 chart !== CHART_TYPE.PIE)
             .forEach(chart => this.chartIntegration.setChartComponentOptions(chart, OPTIONS_TYPE.X_AXIS, { labelAngle: 30 }));
+        
+        this.chartIntegration.onChartTypesDetermined.subscribe((args: IDeterminedChartTypesArgs) => {
+            if (args.chartsAvailability.size === 0 || args.chartsForCreation.length === 0) {
+                this.chartIntegration.disableCharts(this.allCharts);
+            } else {
+                args.chartsAvailability.forEach((isAvailable, chart) => {
+                    if (args.chartsForCreation.indexOf(chart) === -1) {
+                        this.chartIntegration.disableCharts([chart]);
```

Comment:
Am not sure what is the implementation of disable/enableCharts function in chartIntegration package. Possibly it would be more effective and concise, if the method is called once for all charts to enable/disable.